### DISCUSSION TITLE
Adding and improving a CSpell config for content spellchecking

### DISCUSSION THREAD

**Discussion by bsmth** [mdn/mdn-community#437] [Content] [answered]
Hi all, I believe the topic of spellchecking has come up many times - this is difficult because of how many files we have, the number of coding languages we are working with and custom text we use, such as macros. If you run `cspell` over the project without any configuration, the output is completely unmanageable.

I've been using a local cspell config over the last few months, finding appropriate dictionaries (`lorem-ipsum` etc.) occasionally and I think we can slowly start working with it unobtrusively at the authoring stage.

__Proposal for now:__ 
* Check in a cspell config that we can use in IDEs and from the CLI
* Incrementally build out the project dictionary and patterns as we go by accepting some additions to the cspell config in parallel to regular PRs as we work on one or two files. 

__Out of scope:__
* 🚧 spellcheck in CI (this topic is too busy at the moment)  
* 📈 sweeping pull requests in folders across many files
* 🏆 Get down to 0 cspell errors in the next `X` weeks 

I've opened a PR now that can get the ball rolling on this https://github.com/mdn/content/pull/28487/files. It contains:

```bash
.vscode/cspell.json.       # the spellchecker config (dictionaries, ignore patterns)
.vscode/project-words.txt  # custom words we want the spellchecker to ignore
```

We may need to occasionally accept/ignore changes to these two files in content PRs going forward, but I hope that's acceptable.

## Using cspell

From the terminal, it looks like:

```bash
# install if you don't have it
npm install -g cspell@latest

# run against a directory
cspell lint --no-progress --unique --config .vscode/cspell.json ./files/en-us/web/accessibility
```

Looking at IDE support in VSCode, for example, spellchecking and adding unknown words to a project dictionary is quite good via quick-fix (<kbd>command</kbd> + <kbd>.</kbd>)

<img width="544" alt="image" src="https://github.com/mdn/mdn-community/assets/43580235/81a7648d-556f-4e90-95f7-cbf1a9f7a8ab">

Or in the problems panel:

<img width="566" alt="image" src="https://github.com/mdn/mdn-community/assets/43580235/94e5aede-b454-43a5-9e25-54b9cea0ae45">

Let me know what you think!

**Comment by dipikabh** [comment]
Thanks @bsmth. I think this is a good start and a good disclaimer to rule out the things that are out of scope at the moment 👍. At the minimum, we do need to run the spell checker locally (from the CLI or IDE).
Do we need to worry about the impact on translated content as a result of these changes?

**Reply by bsmth** [reply]
>Do we need to worry about the impact on translated content as a result of these changes?

Not that I'm aware of, it should just help us catch typos in `mdn/content` as we're making edits, mostly.

**Comment by bsmth** [comment, marked as answer]
I think we can close as done now. A good summary of activity on this here:

https://github.com/mdn/content/commits/main/.vscode/cspell.json

Thanks, everyone 🎉
